tests/unit/test_charm.py

```diff
@@ -58,59 +76,32 @@ def test_on_config_changes_aborts_if_host_is_container(harness, mocker):
+            "iqn-with-whitespace",
+        ),
+        (INITIATOR_CONTENT_TEMPLATE.format("WrongName=this-is-wrong"), None),
+    ],
+    ids=["no-file", "empty-file", "content-present", "content-with-whitespace", "invalid-content"],
+)
+def test_get_initiator_name_from_file(
+    harness, mocker, tmp_path, initiator_content, expected_initiator_name
+):
     """Test fetching the initiator name from initiatorname.iscsi file."""
-    initiator_content = dedent(
-        """\
-        ###############################################################################
-        # [ WARNING ]
-        # configuration file maintained by Juju
-        # local changes will be overwritten.
-        #
-        # DO NOT EDIT OR REMOVE THIS FILE!
-        # If you remove this file, the iSCSI daemon will not start.
-        # If you change the InitiatorName, existing access control lists
-        # may reject this initiator.  The InitiatorName must be unique
-        # for each iSCSI initiator.  Do NOT duplicate iSCSI InitiatorNames.
-        ###############################################################################
-        InitiatorName=test-iqn"""
-    )
-
-    wrong_initiator_content = dedent(
-        """\
-        ###############################################################################
-        # [ WARNING ]
-        # configuration file maintained by Juju
-        # local changes will be overwritten.
-        #
-        # DO NOT EDIT OR REMOVE THIS FILE!
-        # If you remove this file, the iSCSI daemon will not start.
-        # If you change the InitiatorName, existing access control lists
-        # may reject this initiator.  The InitiatorName must be unique
-        # for each iSCSI initiator.  Do NOT duplicate iSCSI InitiatorNames.
-        ###############################################################################
-        wrongName=test-iqn"""
-    )
-
-    # case 1: File not present
     initiatorname_file = tmp_path / "initiatorname.iscsi"
+    if initiator_content is not None:
+        initiatorname_file.write_text(initiator_content)
     initiator_name = harness.charm._get_initiator_name_from_file(initiatorname_file)
-    assert initiator_name is None
-
-    # case 2: File with empty contents
-    initiatorname_file.write_text("")
-    initiator_name = harness.charm._get_initiator_name_from_file(initiatorname_file)
-    assert initiator_name is None
-
-    # case 3: File with wrong format of "initiatorName"
-    initiatorname_file.write_text(wrong_initiator_content)
-    initiator_name = harness.charm._get_initiator_name_from_file(initiatorname_file)
-    assert initiator_name is None
-
-    # case 4: File with proper contents
-    initiatorname_file.write_text(initiator_content)
-    initiator_name = harness.charm._get_initiator_name_from_file(initiatorname_file)
-    assert initiator_name == "test-iqn"
+    if expected_initiator_name is None:
+        assert initiator_name is expected_initiator_name
```

Comment:
Much much better after use `parametrize`. 

I think no matter the expected_initiator_name is None or not, you can use "==".

```suggestion
        assert initiator_name == expected_initiator_name
```

Reply:
I was keeping the [PEP 8 recommended approach](https://peps.python.org/pep-0008/#programming-recommendations) in mind :smile: 

> Comparisons to singletons like None should always be done with is or is not, never the equality operators.

But on second thought for this specific case, matching for both `""` and `None` works well. I'll make the required changes!